beets/autotag/match.py

```diff
@@ -126,21 +127,15 @@ def assign_items(
     of objects of the two types.
     """
     # Construct the cost matrix.
-    costs: list[list[Distance]] = []
-    for item in items:
-        row = []
-        for track in tracks:
-            row.append(track_distance(item, track))
-        costs.append(row)
-
+    costs = [[float(track_distance(i, t)) for t in tracks] for i in items]
```

Comment:
I'd be curious how much this quadratic steps contributes to the overall runtime (the lap algorithms seem to scale cubically, but the overhead of the Python code in `track_distance` could be significant).

In any case, even if it turns out to matter, looking at the distance computation should probably be a separate PR.

Reply:
While working on this I analysed `track_distance` and `Distance` implementations and found that the only significant overhead was the `config` access.

Reply:
Nice, so you did all the work already :fire: